perf/tools/aggregate.lua

```diff
@@ -0,0 +1,93 @@
+-- File to aggregate the benchmark results from JSON files to the
+-- format parsable by the InfluxDB line protocol [1]:
+-- <measurement>,<tag_set> <field_set> <timestamp>
+--
+-- <tag_set> and <field_set> have the following format:
+-- <key1>=<value1>,<key2>=<value2>
+--
+-- The reported tag set is a set of values that can be used for
+-- filtering data (i.e., branch or benchmark name).
+--
+-- The script accepts the following parameters:
+--
+-- <input_dir> -- the directory from which the .json files are
+--                taken.
+-- <output>    -- the filename where the results are saved.
+--
+-- [1]: https://docs.influxdata.com/influxdb/v1/write_protocols/line_protocol_tutorial/
+
+local json = require('json')
+local fio = require('fio')
+
+local params = require('internal.argparse').parse(arg, {
+    {'input_dir', 'string'},
+    {'output', 'string'},
+})
+
+local input_dir = params.input_dir
+assert(input_dir and fio.path.is_dir(input_dir),
+       'given input_dir is not a directory')
+
+local output = params.output
+local out_fh = assert(io.open(output, 'w'))
+
+local function exec(cmd)
+    return io.popen(cmd):read('*all'):gsub('^%s+', ''):gsub('%s+$', '')
+end
+
+local commit = os.getenv('PERF_COMMIT') or exec('git rev-parse --short HEAD')
+assert(commit, 'can not determine the commit')
+
+local branch = os.getenv('PERF_BRANCH') or
+    exec('git rev-parse --abbrev-ref HEAD')
+assert(branch, 'can not determine the branch')
+
+local tag_set = ('branch=%s'):format(branch)
+
+local function read_all(file)
+    local fh = assert(io.open(file, 'rb'))
+    local content = fh:read('*all')
+    fh:close()
+    return content
+end
+
+local REPORTED_FIELDS = {
+    'cpu_time',
+    'items_per_second',
+    'iterations',
+    'real_time',
+}
+
+local time = os.time()
+
+for _, file in pairs(fio.listdir(input_dir)) do
```

Comment:
I don't get why do you need to traverse a dir and search JSON reports if we know these reports in advance. I would remove a loop here and leave a function that accepts a path to a filename of report. Call `aggregate.lua` in CMake and pass to it paths to all benchmark reports.

Reply:
Traversing them in script is just a little bit more convenient than specifying all paths in the command -- it may lead to a huge command in the future (like we have now with `LUA_CPATH` for our LuaJIT tests). Also, it allows you to easily test the script in the console while updating it.
Ignoring for now.